tutorials/serverless-grafana-with-iap/code/main.tf

```diff
@@ -0,0 +1,282 @@
+locals {
+  apis = ["iam.googleapis.com", "compute.googleapis.com", "run.googleapis.com", "apigateway.googleapis.com", "servicemanagement.googleapis.com", "servicecontrol.googleapis.com", "iap.googleapis.com", "sql-component.googleapis.com", "cloudapis.googleapis.com", "sqladmin.googleapis.com", "secretmanager.googleapis.com"]
+}
+
+data "google_project" "project" {
+  project_id = var.project_id
+}
+
+resource "google_project_service" "project" {
+  for_each = toset(local.apis)
+  project = data.google_project.project.project_id
+  service = each.key
+  disable_on_destroy = false
+}
+
+data "google_compute_default_service_account" "default" {
+  project = data.google_project.project.project_id
+}
+
+resource "google_cloud_run_service" "default" {
+  provider = google-beta
+  name     = "grafana"
+  location = var.region
+  project = data.google_project.project.project_id
+
+  metadata {
+    annotations = {
+      "run.googleapis.com/ingress" : "internal-and-cloud-load-balancing"
+    }
+  }
+  
+  template {
+    spec {
+      service_account_name = data.google_compute_default_service_account.default.email
+      containers {
+        image ="mirror.gcr.io/grafana/grafana:${var.grafana_version}"
+        ports {
+          name = "http1"
+          container_port = 8080
+        }
+        env {
```

Comment:
minor - might improve readibility if we moved all these env vars to a file?

Reply:
That would be great, but I lack the idea how to do that. Any recommendations?